client/components/common/tce-core/MetaInput.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <component
+    :is="component"
+    @update="(key, value) => $emit('update', key, value)"
+    :meta="metaInput"
+    :class="{ required: get(metaInput, 'validate.required') }" />
+</template>
+
+<script>
+import get from 'lodash/get';
+import { getMetaName } from 'tce-core/utils';
+
+// TODO: Remove deprecated types in v5.0
+const getDeprecationWarning = (deprecatedType, type, config) =>
+  `Deprecation notice:
+    '${deprecatedType}' meta type is deprecated and will be removed in v5.0.
+    Please use '${type}' instead with following config added: ${config}.`;
+
+const deprecatedTypes = {
+  DATE: {
+    type: 'DATETIME',
+    config: { hideTime: true },
+    warning: getDeprecationWarning('DATE', 'DATETIME', '{ hideTime: true }')
+  },
+  MULTISELECT: {
+    type: 'SELECT',
+    config: { multiple: true },
+    warning: getDeprecationWarning('MULTISELECT', 'SELECT', '{ multiple: true }')
+  }
+};
+
+export default {
+  name: 'meta-input-wrapper',
+  props: {
+    meta: { type: Object, required: true }
+  },
+  computed: {
+    originalType: vm => (vm.meta.type || '').toUpperCase(),
+    deprecatedType: vm => deprecatedTypes[vm.originalType],
+    type: vm => get(vm.deprecatedType, 'type', vm.originalType),
+    metaInput: ({ meta, deprecatedType }) => {
+      return !deprecatedType ? meta : {
+        ...meta,
+        ...deprecatedType.config
+      };
+    },
+    component: vm => getMetaName(vm.type)
+  },
+  methods: { get },
+  watch: {
+    type: {
```

Comment:
If my understanding is correct, the `type` of this meta input never changes. So there should be no need for a watcher. This logic could get triggered in the `mounted` handler.

Reply:
This is just to guard for all possible uses of meta inputs where if it's not keyed properly can end up just changing `meta` prop so I've added watcher with `immediate` flag instead of just `created` hook.